mesonbuild/compilers.py

```diff
@@ -2042,6 +2042,15 @@ def get_option_link_args(self, options):
             return options['c_winlibs'].value
         return []
 
+    @staticmethod
+    def get_gcc_version(executable):
+        p = subprocess.Popen([executable, '-dumpversion'], stdout=subprocess.PIPE,
```

Comment:
I basically just used the same idea for the --version in the codebase.
Will look into it, thanks.